Librería Cashflows
===

Notas de clase sobre ingeniería economica avanzada usando Python

**Juan David Velásquez Henao**  
jdvelasq@unal.edu.co   
Universidad Nacional de Colombia, Sede Medellín  
Facultad de Minas  
Medellín, Colombia 

**Software utilizado**
>Este es un documento interactivo escrito como un notebook de [Jupyter](http://jupyter.org/) , en el cual se presenta un tutorial sobre finanzas corporativas usando Python. Los notebooks de Jupyter permiten incoporar simultáneamente código, texto, gráficos y ecuaciones. El código presentado en este notebook puede ejecutarse en los sistemas operativos Linux y OS X.  

>Haga click [aquí](https://github.com/jdvelasq/guias-de-instalacion) para obtener instrucciones detalladas sobre como instalar Jupyter en Windows y Mac OS X.

>Descargue la última versión de este documento a su disco duro; luego, carguelo y ejecutelo en línea en Try [Jupyter!](https://try.jupyter.org/)



#### Contenido

>* 

**Bibliografía** 

> * [1] SAS/ETS 14.1 User's Guide, 2015.   
* [2] **hp** 12c platinum financial calculator. User's guide. 
* [3] HP Business Consultant II Owner's manual.
* [4] C.S. Park and G.P. Sharp-Bette. Advanced Engineering Economics. John Wiley & Sons, Inc., 1990.

**`cashflows`** es una librería para la ejecución de cálculos financieros. Las funciones implementadas son similares a las usadas en Microsoft Excel, las calculadoras financieras y otros softwares similares. 

In [2]:
# importa la librería financiera.
# solo es necesario ejecutar la importación una sola vez.
import cashflows as cf

# Función `pvfv`

`pvfv(pval=None, fval=None, nrate=None, nper=None, pyr=1, noprint=True)` 

<img src="images/pvfv.png" width=400>

La función `pvfv` returna el valor faltante en la siguiente ecuación:

$$fval = - pval * \left(1 + \frac{nrate}{pyr}\right) ^ {nper}$$

donde:

* `pval` -- valor presente.

* `fval` -- valor futuro.

* `nper` -- cantidad de periodos.

* `nrate` -- tasa de interés nominal.

* `pyr` -- número de periodos de capitalización por año.

**Ejemplo (cuenta de ahorros).--** [3, pág. 88] Se depositan \$ 2000 en una cuenta de ahorros que paga un interés anual del 7.2% (calculado anualmente). Si no se hacen otros depósitos en la cuenta, cuanto tiempo se requiere para que la cuenta tenga \$ 3000? R/ 5.83

<img src="images/sesion-1-ejemplo-4.png" width=350>

In [7]:
cf.pvfv(nrate =     7.2, # tasa de interes
        pval  = -2000,     # valor presente
        fval  = +3000)     # valor futuro

5.8318433820838607

In [8]:
# Ya que nper es un valor entre 5 y 6, se requieren 6 años 
# para tener un balance de al menos $ 3000.
# El balance al final de los seis años es (R/ 3035.28):
cf.pvfv(nrate =     7.2, # tasa de interes
        pval  = -2000,   # valor presente
        nper  = 6)       # numero de periodos

3035.2796326007801

**Ejemplo.--** Cuál será el valor futuro de \$100, \$ 200, \$ 300 y \$ 400 en 5 años a una tasa de interés de 3% anual? 

In [48]:
cf.pvfv(pval  =  [100, 200, 300, 400], # una de las variables puede ser un vector
        nper  =  5,
        nrate =  3.0)

[-115.92740796091358,
 -231.8548153909136,
 -347.78222282091355,
 -463.70963025091356]

**Ejemplo.--** Cuál será el valor futuro de \$100 en 1, 2, 3 y 4 años a una tasa de interés de 3% anual? 

In [49]:
cf.pvfv(pval  =  100, # una de las variables puede ser un vector
        nper  =    [1, 2, 3, 4],
        nrate =    3.0)

[-103.0000001, -106.09000020299999, -109.27270030909, -112.55088141836272]

# Función `pvpmt`

`pvpmt(pmt=None, pval=None, nrate=None, nper=None, pyr=1, noprint=True)`

<img src="images/pvpmt.png" width=400>

Nomenclatura para los parámetros:

* `pval` -- valor presente.

* `pmt` -- pago periodico.

* `nper` -- cantidad de periodos.

* `nrate` -- tasa de interés por periodo.

* `pyr` -- número de periodos por año.

**Ejemplo.--** [2, pág. 57] Si se va a realizar un leasing a una tasa de interés de (5.9%/12) y se deben realizar 48 pagos de \$ 450 y un pago inicial de \$ 1500 al constituirse el crédito, cuál es el monto del préstamo?

In [23]:
cf.pvpmt(pmt   =  -450,      # pago mensual
         nrate =     5.9/12, # tasa de interés
         nper  =    48)      # numero de periodos

19198.600408593771

In [24]:
_ + 1500

20698.600408593771

**Ejemplo.--** [2, pág. 59] Calcule el pago mensual de una hipoteca por \$ 243400 pagada en 348 meses a una tasa del 5.25%/12.

In [26]:
cf.pvpmt(pval  = 243400,       # monto
         nrate =      5.25/12, # tasa de interés
         nper  =    348)       # numero de periodos

-1363.2868774076235

**Ejemplo.--** [3, pág. 81] Se está financiando la compra de un carro nuevo con un leasing a tres años a una tasa de interés del 10.5%/12. El precio del carro es de \$ 7250. Se debe realizar un pago inicial de \$ 1500. Cuánto es el pago mensual si los pagos se hacen al final del mes? 

In [31]:
cf.pvpmt(pval  = 7250-1500,
         nrate =   10.5/12,   
         nper  =   36)

-186.88905015194925

In [32]:
# si se desea reducir el pago periodico en $ 10, cual tasa de interés debería obtenerse.
cf.pvpmt(pval = 7250-1500,
         pmt  = -176.89,   
         nper =   36)

0.56260172700844047

**Ejemplo.--** [3, pág. 82] El pago máximo mensual por una hipoteca a 30 años es de \$ 630. Si usted puede realizar un pago inicial de \$ 12.000 y la tasa es de 11.5%/12, cuál es el precio máximo de la propiedad?

In [33]:
cf.pvpmt(pmt   =  -630,   
         nrate =    11.5/12,
         nper  = 30*12) + 12000

75617.636073352623

**Ejemplo.--** [3, pág. 83] Se constituye una hipoteca a 25 años por un monto de \$ 75250 y una tasa de interés mensual de 13.8%/12. (a) cuál es el pago mensual; (b) Si se anticipa la venta de la propiedad al final de 4 años, cuánto debe pagarse para cancelar la deuda?   

In [34]:
# (a) R/ $ 894.33
cf.pvpmt(pval = 75250,   
         nrate =   13.8/12,
         nper = 25*12) 

-894.33055797115355

# Función `pmtfv`

`pmtfv(pmt=None, fval=None, nrate=None, nper=None, pyr=1, noprint=True)`

<img src="images/pmtfv.png" width=400>

Nomenclatura para los parámetros:

* `pmt` -- pago periodico.

* `fval` -- valor futuro.

* `nper` -- cantidad de periodos.

* `nrate` -- tasa de interés por periodo.

* `pyr` -- número de periodos por año.

# Función `tvmm`

`tvmm(pval=None, fval=None, pmt=None, nrate=None, nper=None, due=0, pyr=1, noprint=True)`

**`tvm`** es un modelo para la realización de cálculos simples del valor del dinero en el tiempo. Este modelo usa variables internas para almacenar la información y funciones para realizar los cálculos.

Nomenclatura para los parámetros:

* `pval` -- valor presente.

* `fval` -- valor futuro.

* `pmt` -- pago periodico.

* `nper` -- cantidad de periodos.

* `nrate` -- tasa de interés por periodo.

* `pyr` -- número de periodos por año.

* `due` -- momento del periodo en que se paga la anualidad: `'end'` (o 0) indica el pago al final del periodo; `'begin'` (o 1) indica el pago al principio del periodo.

Nomenclatura para las funciones de equivalencia financiera:

# Función `amortize`

`amortize(pval=None, fval=None, pmt=None, nrate=None, nper=None, due=0, pyr=1, noprint=True)`

Imprime la tabla de amortizaciones para los cálculos realizados.

A continuación se presentan varios ejemplos de su uso.

**Ejemplo (hipoteca).--** [2, pág. 50] Se hará un préstamo de \$ 35000 con un interes del (10.5% / 12). Si se hacen pagos mensuales de \$ 325 al final de cada mes, cuánto tiempo se requiere para cancelar la deuda?  

In [14]:
## crea una instancia del modelo y se almacena en una variable
m = cf.tvmm(nrate =    10.5/12, # tasa de interes
             pval = 35000,      # valor presente  
             pmt  =  -325,
             fval =     0,
             due  =     0)      # 'end' 
m

327.43926528279928

Si se hacen 327 pagos de \$ 325, cuánto será el pago No. 328? 

In [15]:
# pago en exceso el valor de la deuda
m = cf.tvmm(nrate =    10.5/12, # tasa de interes
             pval = 35000,      # valor presente  
             pmt  =  -325,
             nper =   327,
             due  =     0)      # 'end' 
m

-141.86862789513543

In [16]:
# pago en exceso el valor de la deuda
m * (1 + 10.5/12/100) + 325

181.89002161078213

Se se hacen únicamente 327 pagos, cuál es el valor del pago final para cancelar completamente la deuda?

In [17]:
m = cf.tvmm(nrate =    10.5/12, # tasa de interes
             pval = 35000,      # valor presente  
             pmt  =  -325,
             nper =   327,
             due  =     0)      # 'end' 
m

-141.86862789513543

In [18]:
-325 + m

-466.86862789513543

**Ejemplo.--** [2, pág. 53] Se abre una cuenta hoy con un depósito de \$ 775. La tasa de interés es de (6.25%/24). Si se siguen realizando depósitos de \$ 50, cuánto tiempo se requiere para alcanzar un saldo de \$ 4000?   

In [19]:
cf.tvmm(nrate =  6.25/24, # tasa de interes
        pval   = -775,    # deposito inicial  
        pmt    =  -50,    # depositos periodicos
        fval   = 4000,    # saldo final
        due    = 0)       # depositos al final del periodo 


57.548385170743252

In [20]:
# saldo al final de 58 periodos
cf.tvmm(nrate  =    6.25/24, # tasa de interes
        nper   =   58,       # numero de periodos
        pval   = -775,       # deposito inicial  
        pmt    =  -50,       # depositos periodicos
        due    = 0)          # depositos al final del periodo 

4027.2656060308091

**Ejemplo.--** [2, pág. 55] Qué tasa de interés debe obtenerse para acumular \$ 10000 en 32 periodos si se hace una inversión de \$ 6000? R/ 1.61%

In [21]:
cf.pvfv(pval = -6000,         # deposito inicial
        nper =    32,         # numero de periodos
        fval = 10000)         # saldo final

1.6091394935432066

In [22]:
cf.tvmm(pval = -6000,     # deposito inicial
        nper =    32,     # numero de periodos
        pmt  = 0,         # pago periodico
        fval = 10000)     # saldo final

1.6091394935432066

**Ejemplo.--** [2, pág. 58] Cuánto se puede pagar por una propiedad que generará un flujo neto anual de \$ 17500 durante 5 años, si al final la propiedad se puede vender en \$ 540.000? (la tasa de interés es del 12%)  

In [25]:
cf.tvmm(pmt   =  17500,   # pago periodico anual
        fval  = 540000,   # valor de venta
        nrate =     12.0, # tasa de interés
        nper  =      5)   # numero de periodos

-369494.08562908118

**Ejemplo.--** [2, pág. 59] Cuánto es el monto periodico que debe consignarse mensualmente en una cuenta de ahorros si el saldo inicial es de \$ 3200, el saldo final es de \$ 60000, la tasa es de 9.75%/2 y plazo es de 30 meses? 

In [27]:
cf.tvmm(pval  = -3200,      # apertura
        fval  = 60000,      # saldo futuro
        nper  =    30,      # numero de periodos
        nrate =     9.75/2) # tasa de interés

-717.44030300066242

**Ejemplo.--** [2, pág. 61] Si se tiene una hipoteca de \$ 243400 con un pago mensual de \$ 1363.29 en 348 meses a una tasa del 5.25%/12, cuál es el pago que debe realizarse en la cuota 60 para cancelar completamente la deuda?

In [28]:
cf.tvmm(pval  = 243400,        # monto
        nrate =      5.25/12,  # tasa de interés
        pmt   =  -1363.29,     # pago mensual
        nper  =     60)        # numero de periodos

-222975.98179576019

**Ejemplo.--** [2, pág. 61] Si se consignan \$ 50 al principio de cada mes en una nueva cuenta que paga 6.25%/12, cuál es el saldo al final de 24 meses?   

In [29]:
cf.tvmm(pval  =   0,        # monto
        nrate =   6.25/12,  # tasa de interés
        pmt   = -50.0,      # pago mensual
        due   =   1,        # pago al principio del periodo
        nper  =  24)        # numero de periodos

1281.3359267030764

**Ejemplo.--** [2, 62] Se compra una propiedad por \$ 32000. Si se presenta una depreciación del 2% por año, cuál será el valor de la propiedad al final de 6 años? R/ \$ 28346.96

In [30]:
cf.pvfv(pval  = -32000,
        nrate =     -2.0,   
        nper  =      6)

28346.956187077209

In [35]:
# (b) R/ $ -73408.81
cf.tvmm(pval  = 75250, 
        pmt   =  -894.33,
        nrate =    13.8/12,
        nper  =  4*12) 

-73408.811623403832

 **Ejemplo (leasing).--** [3, pág 02] Se hace un leasing por una maquinaria por 4 años (48 meses) con pagos mensuales de \$ 2400; se debe pagar una cuota adicional de \$ 2400 al principio del leasing para reemplazar la última cuota (que ocurre al principio del mes 48). El contrato incluye una opción de compra al final del periodo de leasing por \$ 15000. Cual valor capitalizado del leasing para una tasa del 1.5% mensual? 

<img src="images/sesion-1-ejemplo-5.png" width=350>

In [11]:
# se pagan 47 cuotas de $ 2400 al principio del mes.
cf.tvmm(nper  =     47,
        pmt   =  -2400,
        due   =      1,
        fval  =      0,
        nrate =      1.5)
# valor presente de las 47 cuotas al principio del periodo R/ $ 81735.58

81735.575590862747

In [12]:
x = _ + 2400 # + cuota adicional al principio del leasing  R/ $ 84135.58
x

84135.575590862747

In [13]:
x + cf.pvfv(nper  =     48,
            fval  = -15000,
            nrate =      1.5)  # opcion de compra R/ $ 91476.00



91476.001017201139

**Ejemplo (ahorro programado).--** Se abre una cuenta ahorros con un depósito inicial de \$ 2000, y posteriormente se hacen depósitos quincenales de \$ 80 durante 15 años. La cuenta paga un interés del 0.346% quincenal. Cuánto dinero habrá en la cuenta después del último depósito? R/ 63988.44 

In [36]:
cf.tvmm(pval  =   -2000,
        pmt   =     -80,
        nper  = 15*12*2,
        nrate =       0.346)

63988.433987522512

**Ejemplo (leasing).--** Se hace un leasing por 3 años para un carro nuevo que vale hoy \$ 13500, con la opción de comprar el carro por \$ 7500 al final del leasing. Cuánto es el pago mensual si el interés es del 1.16% mensual? (tenga en cuenta que en el leasing se paga la cuota al principio del periodo) R/ $ 288.49

In [37]:
cf.tvmm(pval  =   13500,
        fval  =   -7500,
        nper  =      36,
        due   =       1,
        nrate =       1.16)

-288.4862833217752

**Ejemplo (préstamo).--** Se va a comprar un equipo con un préstamo por \$ 35000 con pagos mensuales de \$ 325 mensuales. Si la tasa de interés es del 0.88% mensual, cuántos periodos se requieren para realizar el pago de la deuda?

In [38]:
x = cf.pvpmt(pval  = 35000,
             nrate =     0.88,
             pmt   =  -325)
x # R/ 336.77

336.76995140884333

In [39]:
# OPCION 1: se hace nper = 337 para calcular cuanto se paga 
# como excedente en la última cuota. R/ $ 74.51
cf.tvmm(pval = 35000,
        pmt   = -325,
        nrate =    0.88,
        nper  =  337)


74.513808694551699

In [40]:
325 - _  # Pago final, restando el excedente.  R/ \$ 250.49

250.4861913054483

In [41]:
# OPCION 2: Se hace un pago adicional con la 
# cuota 336 para cancelar el préstamo
cf.tvmm(pval = 35000,
        pmt   = -325,
        nrate =    0.88,
        nper  =  336)

-248.30114126228727

In [42]:
-325 + _ # pago final: cuota actual + saldo por cancelar

-573.30114126228727

**Ejemplo (inversión).--** Se comprarán unas bodegas para arrendamiento. Si el arriendo es de \$ 17500 anual, y las bodegas se venden al final del año 5 por \$ 540000. Cuál es el monto máximo que puede pagarse por las bodegas si la tasa es del 12% anual?  R/ \$ 369494.09

In [44]:
cf.tvmm(fval  = 540000,
        pmt   =  17500,
        nrate =     12.0,   
        nper  =      5)

-369494.08562908118

**Ejemplo (ahorro programado).--** Si al principio de cada mes se ahorran \$ 100 a una tasa del 2% durante 24 meses, ¿cuánto dinero se tendrá al final del mes 24?.

In [45]:
cf.tvmm(nrate =    2.0,
        pval  =    0,
        nper  =   24,
        pmt   = -100,
        due   =    1)

3103.0299723236531

**Ejemplo (préstamo).--** Se desea determinar el valor de las cuotas mensuales iguales que se deberán pagar por un préstamo de \$ 16700 financiado a 5 años al 2% mensual

In [46]:
cf.pvpmt(nrate =      2.0,
         pval  =  16700,
         nper  =   5*12)

-480.42502929096554

**Ejemplo (préstamo con pago final adicional).--** Se realiza una hipoteca por \$ 243400 a 5 años y con una tasa de interes de 0.12% con pagos mensuales de \$ 4000 y un pago adicional en el último mes. ¿cuál es el valor de dicho pago adicional?

In [47]:
cf.tvmm(nrate =       0.12,
        pval  =  243400,
        nper  =    5*12,
        pmt   =   -4000)

-12863.296612548758

**Ejemplo.--** Cuál será el valor futuro de \$100 si se invierte a un año al 1%, a 2 años al 1.5% y a 3 años al 1.7%, con una capitalización anual? 

In [50]:
cf.tvmm(pval  =  100,               # si varias variables son listas
        pmt   =    0,               # todas ellas deben tener la misma
        nper  =    [1,   2,   3],   # longitud
        nrate =    [1.0, 1.5, 1.7]) # 

[-101.0000001, -103.02250020149998, -105.18719160512887]

**Ejemplo.--** Cuál es la amortización para los siguientes préstamos? (`fv` es el pago final residual (como en el leasing)
```
      plazo    5,        5,       6,       7
      pv     100,      110,     110,     105
      fv     -20,      -10,     -20,       0
      tasa     0.020,    0.017,   0.016,   0.017
      
```

In [51]:
cf.tvmm(pval  = [ 100,   110,   110,   105   ],
        fval  = [ -20,   -10,   -20,     0   ],         
        nper  = [   5,     5,     6,     7   ], 
        nrate = [   2.0,   1.7,   1.6,   1.7 ])

[-17.372671528345773,
 -21.201461428344242,
 -16.1711096183316,
 -16.037190189122793]

**Ejemplo (tablas de amortización - Parte 1).--** Construya la tabla de amortización (balance) para un préstamo de  \$ 1000 a 6 meses con pagos mensuales iguales a una tasa de interés del 1% mensual. 

In [52]:
# principal, interest, payment, balance
ppal, interest, payment, balance = cf.amortize(pval=1000, fval=0, pmt=None, nrate=1.0, nper=6, due=0)

In [53]:
ppal

[0,
 -162.54836671088106,
 -164.17385037798988,
 -165.81558888176977,
 -167.47374477058747,
 -169.14848221829334,
 -170.83996704047627]

In [54]:
sum(ppal)

-999.99999999999773

In [55]:
interest

[0,
 10.0,
 8.37451633289119,
 6.7327778291112912,
 5.0746219402935928,
 3.3998844925877183,
 1.7083996704047848]

In [56]:
sum(interest)

35.290200265288583

In [57]:
payment

[0,
 -172.54836671088106,
 -172.54836671088106,
 -172.54836671088106,
 -172.54836671088106,
 -172.54836671088106,
 -172.54836671088106]

In [58]:
sum(payment)

-1035.2902002652863

---

# Ejercicios

**Ejercicio.--** Suponga que usted quiere asegurarse de que puede financiar la educación universitaria de su hija dentro de 14 años a partir de hoy. Su hija retirará \$ 500 al inicio de cada mes de una cuenta de ahorros durante 4 años. ¿Cuánto se tiene que depositar hoy en la cuenta si el interés es del 1% mensual?

**Ejercicio.--** Mirando hacia la jubilación, usted desea acumular \$ 60000 después de 15 años haciendo depósitos en una cuenta que paga 5.3% de interés semestral. Usted abre la cuenta con un depósito de \$ 3200 con la intención de hacer depósitos semestrales, comenzando seis meses más tarde a partir de sus ingresos. Calcular la cantidad de dinero que se deberá depositar semestralmente (no se consigna al final del último semestre). 

---

In [1]:
import cashflows as cf

# Conversión de tasas de interés

## Interés anticipado e interés vencido

[Contenido](#Contenido)

<img src="images/antxven.png" width=500>

**Interés vencido:** se paga al final del periodo.  

$$F=P(1+r)$$

**Interés anticipado:** se paga al inicio del periodo (antes de su causación). En este caso surge una paradoja, que el interés se puede reinvertir a la misma tasa de interés (anticipadamente):

$$F = P + Pr_*  + Pr_*^2 + ...$$

La suma infinita anterior puede reescribirse como:

$$F=P \frac{1}{1 - r_*}$$

Igualando las dos ecuaciones anteriores:

$$\frac{1}{1 - r_*}=1+r ~,~~ r=\frac{r_*}{1 - r_*} ~,~~ r_*=\frac{r}{1 + r}$$  

Interés efectivo anual: 

$$r_a=\left[1+\left (\frac{r_*}{1-r_*}\right)\right]^n-1$$

**`cashflow`** tiene las funciones `nom2eff()` y `eff2nom()` para realizar las conversiones entre nominal (anticipado y vencido) y efectivo respectivamente.

**Ejemplo.--** Se solicita un prestamo a un año con un interés anticipado del 20%. ¿Determine el interés efectivo pagado por el dinero?

In [2]:
0.2 / (1 - 0.2)

0.25

**Ejemplo.--** Si se desea obtener una tasa efectiva anual del 36%, ¿cuánto se deberá cobrar en forma anticipada anual para obenerla?

In [3]:
0.36 / (1 + 0.36)

0.2647058823529412

## Interés nominal e interés efectivo

[Contenido](#Contenido)

**Interés nominal** (r): expresado sobre una base anual para un número M de periodos de pago en el año. 

**Interés efectivo por periodo de pago **($i$): representa el interés real para cada periodo de pago en el año.  

**Interés efectivo anual** ($i_a$): interés real para un periodo único de pago de un año.  

$$ i= \frac{r}{M},$$  
$$i_\alpha = \left( \displaystyle 1 + \frac{r}{M}\right)^M - 1 $$

**Ejemplo.--** Se está considerando abrir una cuenta de ahorros en uno de tres bancos. Cuál banco tienen la tasa de interés más favorable?

* Banco #1: 6.72% anual, compuesto semestralmente

* Banco #1: 6.70% anual, compuesto trimestralmente.

* Banco #2: 6.65% anual, compuesto mensualmente.

In [4]:
cf.iconv(nrate = 6.72, pyr =  2)  ## Banco 1

(6.8328960000000105, 3.36)

In [5]:
cf.iconv(nrate = 6.70, pyr =  4)  ## Banco 2 -- mejor opción 

(6.8702251402816605, 1.675)

In [6]:
cf.iconv(nrate = 6.65, pyr = 12)  ## Banco 3

(6.85647762811652, 0.5541666666666667)

In [7]:
## Otra forma
cf.iconv(nrate = [6.72, 6.79, 6.65], pyr = [2, 4, 12])

([6.8328960000000105, 6.964855220821997, 6.85647762811652],
 [3.36, 1.6975, 0.5541666666666667])

**Ejemplo.--** Convierta una tasa del 12% anual compuesto semestralmente a anual compuesto mensualmente.

In [8]:
erate, _ = cf.iconv(nrate = 12.0, pyr = 2) ## efectiva por año 
erate

12.360000000000015

In [9]:
nrate, _ = cf.iconv(erate = erate, pyr = 12)       ## nominal compuesta mensualmente
nrate

11.710553015030811

**Ejemplo.--** Sea un interés nominal del 12% capitalizado mensualmente. Calcule:

* Tasa efectiva mensual
* Tasa efectiva trimestral
* Tasa efectiva anual

<img src="images/tasa-nominal-efectiva.png" width=600>

In [10]:
## tasa efectiva mensual
0.12 / 12

0.01

In [11]:
## tasa efectiva trimestral
erate, _ = cf.iconv(nrate = 3 * 0.12 / 12, pyr = 3)
erate

0.030003000099987354

In [12]:
## tasa efectiva anual
erate, _ = cf.iconv(nrate = 12.0, pyr = 12)
erate

12.682503013196978

## Nomenclatura

[Contenido](#Contenido)

<img src="images/nomenclatura.png" width=600>

---

## Ejercicios

**Ejercicio.--** Cuál es la tasa efectiva anual equivalente a 15% N.A.M.V. (nominal anual mes vencido)?

**Ejercicio.--** Cuál es la tasa efectiva anual equivalente a 23% N.A.T.A. (nominal anual trimestre anticipado)?

**Ejercicio.--** Sea un interés nominal del 39.29% capitalizado mensualmente a cuánto equivale en términos semestrales? (R/ 6.15%)

**Ejercicio.--** ¿Cuál es el valor futuro de \$ 609 dentro de 2 años a una tasa del 2% NATV? 

**Ejercicio.--** ¿Cuál es el valor presente de un pago único de \$ 890 recibido dentro de 6 años a una tasa de 2.7% NATA?   

**Ejercicio.--** ¿Qué cantidad de dinero se poseerá después de prestar \$ 2300 al 27% NAMA durante 3 años?

**Ejercicio.--** ¿Cuál es la tasa efectiva semestral equivalente al 14% NMTA?

**Ejercicio.--** ¿Cuánto dinero mensual se debe empezar a abonar hoy si se desea reunir \$ 28700 al final de 5 años y los ahorros rentan el 16%?

**Ejercicio.--**  Se decide ahorrar mensualmente \$ 900 los cuales depositará al principio de cada mes en una entidad financiera que paga un interés del 30%. ¿Cuánto habrá acumulado al cabo de 2 años?

---

[Contenido](#Contenido)